core/trino-main/src/main/java/io/trino/sql/planner/plan/JoinNode.java

```diff
@@ -357,7 +358,7 @@ public JoinNode withReorderJoinStatsAndCost(PlanNodeStatsAndCostSummary statsAnd
 
     public boolean isCrossJoin()
     {
-        return criteria.isEmpty() && filter.isEmpty() && type == INNER;
+        return criteria.isEmpty() && filter.orElse(TRUE_LITERAL).equals(TRUE_LITERAL) && type == INNER;
```

Comment:
I don't think we want this. This raises more questions than answers: "should an expression like `1 = 1` or `true and true` be considered a cross join?

The point of this method is to capture a type of join that has no filter and no conditions. Ideally, we'd have a separate node type for this, but that's a much larger topic.

A join with `true` as a condition can be transformed into a cross join (i.e., by removing the filter)